src/cppNGS/FileLocationProviderLocal.cpp

```diff
@@ -0,0 +1,349 @@
+#include "FileLocationProviderLocal.h"
+
+FileLocationProviderLocal::FileLocationProviderLocal(QString gsvar_file, const SampleHeaderInfo header_info, const AnalysisType analysis_type)
+  : gsvar_file_(gsvar_file)
+  , header_info_(header_info)
+  , analysis_type_(analysis_type)
+{
+}
+
+void FileLocationProviderLocal::setIsFoundFlag(FileLocation& file)
```

Comment:
Replace by function that adds the file to the output and checks if it exists:
static addToList(const FileLocation& loc, QList<FileLocation>& list)
{
  list << loc; 
  list.last().is_found = QFile::exists(loc.filename);
}